clang/docs/ReleaseNotes.rst

```diff
@@ -571,6 +571,9 @@ New features
   if class of allocation and deallocation function mismatches.
   `Documentation <https://clang.llvm.org/docs/analyzer/checkers.html#unix-mismatcheddeallocator-c-c>`__.
 
+- Function effects (the ``nonblocking`` and ``nonallocating`` "performance constraint" attributes)
```

Comment:
I would like some elaboration here, reading this after coming in 'cold' I have no idea what this means.  Also, why double-tilde's?  I think we only use 1 for these?  Or is .rst different enough again...?

Reply:
RST needs double backticks for code, yeah. And the plan here is that we’ll add a link to the documentation (#109855) in a later pr (a bit like the release note above). Explaining what all of this is about is too much for a release note imo.

Reply:
Thanks for the help on the backticks.  I think we could probably have the release notes still have a bit more detail.  I don't need a full document, but at least a good sentence or of "here is what this is".

Reply:
I think even that is a little 'requires existing knowledge'?  Basically, the audience for this is 'random redditor trying to figure out what features to use'.  So we need to make sure anything like this is sufficiently titillating while not being so long as to lose interest.  

For most bugs in release notes, we can assume that if the person cares about it, they know the details about the bug, but for features this is sort of our first chance to 'advertise' them.  So I'd like something less vague/that gives some good 'hook' as to why the person wants to use it.

Perhaps I'll make @shafik come in, read the documentation enough to understand it, and propose some text :)

Reply:
I took a shot at something more "hook-y".